### PR TITLE
[DygraphToStatic]Add cast transform for dygraph_to_static.

### PR DESCRIPTION
**Pull request by wzzju**
### PR types
New features

### PR changes
Others

### Describe

Add the type casting transform for `dygraph_to_static`.

### CODE REVIEW

File: python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 from paddle.fluid.dygraph.dygraph_to_static.print_transformer import PrintTransformer
 from paddle.fluid.dygraph.dygraph_to_static.return_transformer import ReturnTransformer
 from paddle.fluid.dygraph.dygraph_to_static.tensor_shape_transformer import TensorShapeTransformer
+from paddle.fluid.dygraph.dygraph_to_static.cast_transformer import CastTransformer
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py]: Here we import with alphabetical order, so please put this line after CallTransformer

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cast.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+@declarative
+def test_mix_cast(x):
+    x = fluid.dygraph.to_variable(x)
+    x = int(x)
+    x = float(x)
+    x = bool(x)
+    x = float(x)
+    return x
+
```
Comment (zhhsplendid) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_cast.py:61]: Add a test case that `x` is not a variable?

File: python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py (hunk 1 above)
Comment (wzzju) [reply python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py]: Done.

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_cast.py (hunk 2 above)
Comment (wzzju) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_cast.py:61]: Done.

File: python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py [hunk 3]
```diff
@@ -21,9 +21,10 @@
 import gast
 
 from paddle.fluid.dygraph.dygraph_to_static.assert_transformer import AssertTransformer
-from paddle.fluid.dygraph.dygraph_to_static.call_transformer import CallTransformer
 from paddle.fluid.dygraph.dygraph_to_static.basic_api_transformer import BasicApiTransformer
 from paddle.fluid.dygraph.dygraph_to_static.break_continue_transformer import BreakContinueTransformer
+from paddle.fluid.dygraph.dygraph_to_static.call_transformer import CallTransformer
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py:26]: Thank you for also correcting CallTransformer!

**Review by zhhsplendid — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py (hunk 3 above)
Comment (wzzju) [reply python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py:26]: Thx😄

**Pull request merged**